charts/tezos-proto-cruncher/scripts/proto-cruncher.py

```
@@ -0,0 +1,60 @@
+import hashlib
+import random
+import base58
+import string
+import os
+import sys
+import re
+
+
+proto_file = sys.argv[1]
+
+def tb(l):
+    return b''.join(map(lambda x: x.to_bytes(1, 'big'), l))
+proto_prefix = tb([2, 170])
+
+PROTO_NAME = os.getenv("PROTO_NAME")
+VANITY_STRING = os.getenv("VANITY_STRING")
+BUCKET_NAME = os.getenv("BUCKET_NAME")
+BUCKET_ENDPOINT_URL = os.getenv("BUCKET_ENDPOINT_URL")
+BUCKET_REGION = os.getenv("BUCKET_REGION")
```

fail fast if vars not set?

---

as per the medium article, users are free to download and run this script on their computer as well. So I'm not going to fail if the s3 variables are not set. For the `VANITY_STRING`, sure.

---

ah, i see. very good